onnxruntime/core/framework/prepacked_weights_container.h

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <unordered_map>
+#include <unordered_set>
+#include <string>
+#include <cstdint>
+
+#include "core/framework/buffer_deleter.h"
+
+#include "core/framework/allocator.h"
+#include "core/platform/ort_mutex.h"
+#include "prepacked_weights.h"
+
+namespace onnxruntime {
+
+class PrepackedWeightsContainer final {
+ public:
+  PrepackedWeightsContainer() {
+  }
+
+  ~PrepackedWeightsContainer() = default;
+
+  AllocatorPtr GetAllocator(const std::string& device_name);
+
+  const PrePackedWeights& GetCachedWeight(const std::string& key);
+
+  void WriteCachedWeight(const std::string& key, PrePackedWeights&& packed_weight);
```

Comment:
nit: You're writing the weight to the cache. So, technically it's not 'cached' yet. May be just GetWeight(), WriteWeight() and HasWeight()?

Reply:
done